Python: Getting Started sanic (#7827)
getsentry/sentry-docs: src/platforms/python/guides/sanic/index.mdx

description: "Learn about using Sentry with Sanic."
---

> The Sanic integration adds support for the [Sanic Web Framework](https://sanicframework.org).

Reviewer: It looks like this is the new URL

```suggestion
The Sanic integration adds support for the [Sanic Web Framework](https://sanic.dev/).
```

## Configure

> If you have the `sanic` package in your dependencies, the Bottle integration will be enabled automatically. There is nothing to do for you except initializing the Sentry SDK.

Reviewer: ```suggestion
If you have the `sanic` package in your dependencies, the Bottle integration will be enabled automatically when you initialize the Sentry SDK.
```

<SignInNote />

```python
from sanic import Sanic
import sentry_sdk

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    traces_sample_rate=1.0,
)

app = Sanic(__name__)
```

## Verify

```python
from sanic import Sanic
from sanic.response import text

sentry_sdk.init(...)  # same as above

app = Sanic(__name__)

@app.get("/")
async def hello_world(request):
    1 / 0  # raises an error
    return text("Hello, world.")`
```

> When you point your browser to [http://localhost:8000/](http://localhost:8000/) a error event will be sent to [sentry.io](https://sentry.io).

Reviewer: ```suggestion
When you point your browser to [http://localhost:8000/](http://localhost:8000/) an error event will be sent to [sentry.io](https://sentry.io).
```
